chartfx-generate/src/main/java/de/gsi/generate/CodeGenerator.java

```diff
@@ -0,0 +1,250 @@
+import static java.nio.file.StandardOpenOption.*;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.file.*;
+import java.util.List;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * Proof of concept code generator that gets called by the maven exec plugin during the generate
+ * sources phase. It checks the source files for code gen templates by assuming some naming convention,
+ * and then does search/replace based on sections within the source file. This allows multiple
+ * output types to be generated into a single class.
+ *
+ * Currently the generated class inherits from the base class, but I'm not sure that's necessary.
+ * We could just generate a full copy of the base class and then add sections within the template
+ * area. TBD.
+ *
+ * @author ennerf
+ * @author Alexander Krimm
+ */
+@Mojo(name = "generate-sources", defaultPhase = LifecyclePhase.GENERATE_SOURCES)
+public class CodeGenerator extends AbstractMojo {
+    public static final String PROTOTYPE_TYPE = "double";
+    public static final String PROTOTYPE_CLASS_SUFFIX = "DoubleProto";
+    public static final String JAVA_FILE_SUFFIX = ".java";
+    public static final String PROTOTYPE_TYPE_BOXED = "Double";
+    public static final String AUTOGENERATED_WARNING_HEADER = "// This file has been generated automatically. Do not modify!\n";
+    final String[] outputTypes = { "float", "char", "short", "int", "long", "boolean", "byte" };
+
+    @Parameter(defaultValue = "${project}", required = true, readonly = true)
+    MavenProject project;
+
+    @Parameter(defaultValue = "${project.build.sourceDirectory}", readonly = true)
+    String input;
+    @Parameter(defaultValue = "${project.build.directory}/generated-sources/codegen", readonly = true)
+    String output;
+
+    @Override
+    public void execute() throws MojoExecutionException {
+        final Path inputPath = Path.of(this.input);
+        final Path outputPath = Path.of(this.output);
+
+        project.addCompileSourceRoot(this.output); // Add the generated classes to the build path
+        if (getLog().isInfoEnabled()) {
+            getLog().info("Added directory for generated sources to compile sources: " + outputPath);
+        }
+
+        // Adding other types to helper function classes (...GenBase.java)
+        extendUtilityClass(inputPath, outputPath);
+
+        // Adding new classes for different types (...DoubleProto.");java)
+        addDataTypeVariants(inputPath, outputPath);
+    }
+
+    private void addDataTypeVariants(final Path inputPath, final Path outputPath) throws MojoExecutionException {
```

Comment:
Codacy found an issue: [The method 'addDataTypeVariants(Path, Path)' has an NPath complexity of 2002, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5780327)